src/app/static/src/app/components/plots/bubble/utils.ts

```diff
@@ -0,0 +1,106 @@
+import {BubbleIndicatorValuesDict, Dict, Filter, IndicatorValuesDict, NumericRange} from "../../../types";
+import {getColor} from "../../../store/filteredData/utils";
+import {Feature} from "geojson";
+import {ChoroplethIndicatorMetadata} from "../../../generated";
+import Choropleth from "../Choropleth.vue";
+
+export const toIndicatorNameLookup = (array: ChoroplethIndicatorMetadata[]) =>
+    array.reduce((obj, current) => {
+        obj[current.indicator] = current.name;
+        return obj
+    }, {} as Dict<string>);
+
+const iterateDataValues = function(
+    data: any,
+    indicatorsMeta: ChoroplethIndicatorMetadata[],
+    selectedAreaIds: string[] | null,
+    func: (areaId: string,
+           indicatorMeta: ChoroplethIndicatorMetadata, value: number) => void) {
+
+    for (const row of data) {
+        //TODO: exclude rows based on filters
+        //if (excludeRow(row, selectedRegionFilters)) {
+        //    continue;
+        //}
+
+        const areaId: string = row.area_id;
+
+        if (selectedAreaIds && !selectedAreaIds.includes(areaId)) {
+            continue;
+        }
+
+        for (const metadata of indicatorsMeta) {
+
+            if (metadata.indicator_column && metadata.indicator_value != row[metadata.indicator_column]) {
+                //This data is in long format, and the indicator column's value does not match that for this indicator
+                continue;
+            }
+
+            if (!row[metadata.value_column]) {
+                //No value for this indicator in this row
+                continue;
+            }
+
+            const value = row[metadata.value_column];
+
+            func(areaId, metadata, value);
+        }
+    }
+};
+
+export const getIndicatorRanges = function(data: any,
```

Comment:
So in the next branch this should be easy to adapt to either make use of a set of filters or not when calculating indicator ranges - we'll be adding filters as a parameter to the iterateDataValues method for calculating feature values, and we can either provide filters or not when getting indicator ranges.